mock.go

```diff
@@ -45,12 +49,30 @@ func (m *TerminalMock) SetSize(w, h int) {
 	m.cells = make([]*cell, w*h)
 }
 
+func (m *TerminalMock) SetSizeV2(w, h int) {
```

Comment:
Could you switch the implementation of `SetSize` based on a `bool` field such as `m.v2` instead of defining `SetSizeV2`? 

For example:

``` go
// SetSize changes the pseudo-size of the window.
// Note that SetSize resets added cells.
func (m *TerminalMock) SetSize(w, h int) {
	if m.v2 {
		m.simScreen.SetSize(w, h)
		return
	}

	m.sizeMu.Lock()
	defer m.sizeMu.Unlock()
	m.cellsMu.Lock()
	defer m.cellsMu.Unlock()
	m.width = w
	m.height = h
	m.cells = make([]*cell, w*h)
}
```